pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,699 @@
+
+    @padding_cells.setter
+    def padding_cells(self, val):
+        self.__oarray.SetPropValue("Padding", val)
+
+    @property
+    def coordinate_system(self):
+        """Coordinate system name."""
+        cs_dict = self.__map_coordinate_system_to_id()
+        res = "Global"
+        for name, cs_id in cs_dict.items():
+            if cs_id == self.__cs_id:
+                res = name
+        if res == "Global":
+            self.logger.warning("Coordinate system is not loaded. Save the project.")
+        return res
+
+    @coordinate_system.setter
+    def coordinate_system(self, name):
+        cs_dict = self.__map_coordinate_system_to_id()
+        if name not in cs_dict:
+            self.logger.warning("Coordinate system is not loaded. Save the project.")
+        else:
+            self.__cs_id = cs_dict[name]
+            self.edit_array()
+
+    @pyaedt_function_handler()
+    def delete(self):
+        """Delete the component array.
+
+        References
+        ----------
+
+        >>> oModule.DeleteArray
+
+        """
+        self.__app.omodelsetup.DeleteArray()
+        del self.__app.component_array[self.name]
+        self.__app.component_array_names = list(self.__app.get_oo_name(self.__app.odesign, "Model"))
+
+    @pyaedt_function_handler()
+    def export_array_info(self, array_path=None):  # pragma: no cover
+        """Export array information to a CSV file.
+
+        Returns
+        -------
+        str
+           Path of the CSV file.
+
+        References
+        ----------
+
+        >>> oModule.ExportArray
+
+        """
+        if self.__app.settings.aedt_version < "2024.1":  # pragma: no cover
+            self.logger.warning("This feature is not available in {}.".format(str(self.__app.settings.aedt_version)))
+            return False
+
+        if not array_path:  # pragma: no cover
+            array_path = os.path.join(self.__app.toolkit_directory, "array_info.csv")
+        self.__app.omodelsetup.ExportArray(self.name, array_path)
+        return array_path
+
+    @pyaedt_function_handler()
+    def parse_array_info_from_csv(self, csv_file):  # pragma: no cover
+        """Parse component array information from the CSV file.
+
+        Parameters
+        ----------
+        csv_file : str
+             Name of the CSV file.
+
+        Returns
+        -------
+        dict
+           Ordered dictionary of the properties of the component array.
+
+        Examples
+        --------
+        >>> from pyaedt import Hfss
+        >>> aedtapp = Hfss(projectname="Array.aedt")
+        >>> array_names = aedtapp.component_array_names[0]
+        >>> array = aedtapp.component_array[array_names[0]]
+        >>> array_csv = array.export_array_info()
+        >>> array_info = array.array_info_parser(array_csv)
+        """
+
+        info = read_csv(csv_file)
+        if not info:
+            self.logger.error("Data from CSV file is not loaded.")
+            return False
+
+        components = []
```

Comment:
maybe you need to change this in a dictionary as well. Like in __get_properties_from_aedt()